administrator/components/com_content/controllers/article.php

```diff
@@ -145,6 +146,30 @@ public function batch($model = null)
 	}
 
 	/**
+	 * Method to generate and store share token.
+	 *
+	 * @return  boolean   True if token successfully stored, false otherwise and internal error is set.
+	 *
+	 * @since   3.7
+	 */
+	public function shareDraft()
+	{
+		if (!JSession::checkToken('get'))
+		{
+			$this->app->enqueueMessage(JText::_('JINVALID_TOKEN'));
+			$this->app->redirect('index.php');
+		}
+
+		$title = $this->input->getArray(array('jform' => array('title' => 'string')));
```

Comment:
This code from $title until $return should be wrapped in a try/catch so we can catch any errors generated in the model.